plugins/elantp/fu-elantp-hid-haptic-device.c

```diff
@@ -170,20 +173,30 @@ fu_elantp_hid_haptic_device_ensure_eeprom_iap_ctrl(FuDevice *parent, FuElantpHid
 						sizeof(buf), 
 						error)) {
 		g_prefix_error(error, "failed to read IAPControl: ");
-		return -1;
+		*value = -1;
+		return FALSE;
 	}
 	self->iap_ctrl = fu_memread_uint16(buf, G_LITTLE_ENDIAN);
 
 	if ((self->iap_ctrl & 0x800) != 0x800) {
-        	g_prefix_error(error, "bit11 fail: ");
-		return -2;
+        	g_set_error(error,
+			FWUPD_ERROR,
+			FWUPD_ERROR_READ,
+			"bit11 fail");
+        	*value = -2;
```

Comment:
rather than using the &value to return information, could you use the GError code? e.g. G_IO_ERROR, G_IO_ERROR_PARTIAL_INPUT and then match that condition with g_error_matches() in the caller?